pkg/infra/iac2/resources.go

```diff
@@ -35,28 +33,28 @@ type (
 	}
 )
 
-func (e *KubernetesProvider) Provider() string {
-	return "pulumi"
-}
-
 func (e *KubernetesProvider) KlothoConstructRef() []core.AnnotationKey {
 	return e.ConstructsRef
 }
 
-func (e *KubernetesProvider) Id() string {
-	return fmt.Sprintf("%s:%s:%s", e.Provider(), "kubernetes_provider", e.Name)
-}
-
-func (e *RouteTableAssociation) Provider() string {
-	return "pulumi"
+func (e *KubernetesProvider) Id() core.ResourceId {
+	return core.ResourceId{
+		Provider: "pulumi",
```

Comment:
While we're here, can we pull this to a `const`?

Reply:
im a little confused why the provider is pulumi for id function of the kubernetes provider here

Reply:
oh this is in the pulumi plugin ok

Reply:
I kept them all the same /shrug